### ISSUE TITLE
Warning: Could not map element with type "tag"

### ISSUE THREAD

**Issue by finite2** [jerolimov/react-showdown#62]
We recently switched from npm to yarn to handle our install. Upon switching over, the Markdown components would not longer render and would display a warning:

`Warning: Could not map element with type "tag"`

I have tracked down that this occurs when checking the node types (`node instanceof Element`) [MarkdownView.tsx#L48](https://github.com/jerolimov/react-showdown/blob/master/src/MarkdownView.tsx#L48).

Removing this typecheck along with similar typechecks within this file resolves this issue. I believe this is being caused by a miss matched version of `domhandler`. `domhandler` is an undocumented dependency for `react-showdown`. It is also a dependency of `htmlparser2` (a documented dependency for `react-showdown`.

It might well be more complicated than that because `htmlparser2@4.1` (now on version 6) seems to rely on multiple versions of `domhandler` versions 3 (directly) and 4 (via `domutils@2.0`) .

**Comment by jerolimov** [comment]
Good analyse, any idea to solve this? :)

Maybe its time to update the dependencies again, run the tests and release a new version?

**Comment by jerolimov** [comment]
Maybe its also a good idea to add a domhandler dependency directly to ensure that its the right version. I can check this at the weekend.

**Comment by zero734kr** [comment]
Hello, any updates here? I'm having the same problem

**Comment by n1ru4l** [comment]
```
npm why domhandler
```

shows the conflicting versions. The solution is to force only the installation of one of those, which is `domhandler@4.0.0`, trying to enforce `domhandler@3.3.0` will result in some cannot read children of undefined error.

You can enforce the version locally with yarn resolutions.
Since I am using npm in my project together with vite as a bundler (I did not experience this issue with create-react-app/webpack), and there is no equivalent I had to use the resolve map in the `vite.config.js`:

```js
import { defineConfig } from "vite";
import reactRefresh from "@vitejs/plugin-react-refresh";
import macrosPlugin from "vite-plugin-babel-macros";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [reactRefresh(), macrosPlugin()],
  server: {
    port: 3001,
    proxy: {
      "/api": "http://localhost:3000",
      "/files": "http://localhost:3000",
    },
  },
  resolve: {
    alias: {
      domhandler: "node_modules/domutils/node_modules/domhandler/lib/index.js",
    },
  },
});

```

**Comment by zero734kr** [comment]
@n1ru4l I was having this issue when use react-showdown with Next.js

**Comment by zero734kr** [comment]
https://nextjs.org/docs/advanced-features/module-path-aliases
It seems there is a solution for module path aliases for next.js too. I’ll try this later and share the experience

**Comment by wonderlandpark** [comment]
+1 
I'm having the same issue too.

**Comment by n1ru4l** [comment]
I created https://github.com/jerolimov/react-showdown/pull/65 :)

**Comment by jerolimov** [comment]
Thanks for this fix. It is released as version `2.3.0`.
